crates/red_knot_python_semantic/resources/mdtest/narrow/conditionals/nested.md

```diff
@@ -45,3 +45,46 @@ def _(flag1: bool, flag2: bool):
         # TODO should be Never
         reveal_type(x)  # revealed: Literal[1, 2, 3]
 ```
+
+## Cross-scope narrowing
+
+```py
+from typing import Literal
+
+def f(x: str | None):
+    def _():
+        if x is not None:
+            reveal_type(x)  # revealed: str
+
+    class C:
+        if x is not None:
+            reveal_type(x)  # revealed: str
+
+    # TODO: should be str
+    [reveal_type(x) for _ in range(1) if x is not None]  # revealed: str | None
```

Comment:
The limitation here is that we don't do narrowing for `if` clauses in comprehensions?

At first I thought it was that we don't support comprehensions at all, but I think that's not right, since I see a comprehension case below; maybe the comment can clarify?